src/unix/stream.c

```diff
@@ -1398,6 +1400,42 @@ int uv_write2(uv_write_t* req,
 }
 
 
+size_t uv_write_nwritten(const uv_write_t* req) {
+  return req->write_extra.nwritten;
+}
+
+
+int uv__write_cancel(uv_write_t* req) {
+  uv_stream_t* stream;
+  size_t remaining;
+
+  stream = req->handle;
+
+  /* Already completed, return success (allows calling cancel multiple times) */
+  if (!uv__queue_contains(&stream->write_queue, &req->queue))
+    return 0;
```

Comment:
Shouldn't we return some error if you fail to cancel? Or is the thinking that this is a "try and cancel" kinda operation?

Reply:
This was requested by @bnoordhuis above, but regardless, I independently think that it's not particularly useful to return an error code for a request that is already succeeded (because the effect is the same - the write callback will issue as soon as the loop runs again). There could possibly be protection against cancelling twice, but the other requests don't have it, so I'm not sure it's useful.